purchase_deposit/tests/test_purchase_deposit.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2019 Elico Corp, Dominique K. <dominique.k@elico-corp.com.sg>
+# Copyright 2019 Ecosoft Co., Ltd., Kitti U. <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields
+from odoo.exceptions import UserError
+from odoo.tests.common import Form, TransactionCase
+
+
+class TestPurchaseDeposit(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_model = cls.env["product.product"]
+        cls.account_model = cls.env["account.account"]
+        cls.invoice_model = cls.env["account.move"]
+
+        # Create Deposit Account
+        cls.account_deposit = cls.account_model.create(
+            {
+                "name": "Purchase Deposit",
+                "code": "11620",
+                "account_type": "asset_current",
+            }
+        )
+        # Create products:
+        p1 = cls.product1 = cls.product_model.create(
+            {
+                "name": "Test Product 1",
+                "type": "service",
+                "default_code": "PROD1",
+                "purchase_method": "purchase",
+            }
+        )
+
+        cls.po = cls.env["purchase.order"].create(
+            {
+                "partner_id": cls.env.ref("base.res_partner_3").id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": p1.id,
+                            "product_uom": p1.uom_id.id,
+                            "name": p1.name,
+                            "price_unit": 100.0,
+                            "date_planned": fields.Datetime.now(),
+                            "product_qty": 42.0,
+                        },
+                    )
+                ],
+            }
+        )
+
+        cls.tax = cls.env["account.tax"].create(
+            {
+                "name": "Tax 20",
+                "type_tax_use": "purchase",
+                "amount": 20,
+            }
+        )
+
+    def test_create_deposit_invoice(self):
+        self.assertEqual(len(self.po.order_line), 1)
+        # We create invoice from expense
+        ctx = {
+            "active_id": self.po.id,
+            "active_ids": [self.po.id],
+            "active_model": "purchase.order",
+            "create_bills": True,
+        }
+        CreateDeposit = self.env["purchase.advance.payment.inv"]
+        self.po.button_confirm()
+        with Form(CreateDeposit.with_context(**ctx)) as f:
+            f.advance_payment_method = "percentage"
+        wizard = f.save()
+        wizard.amount = 10.0  # 10%
+        wizard.deposit_account_id = self.account_deposit
+        wizard.create_invoices()
+        # New Purchase Deposit is created automatically
+        deposit_id = self.env.company.purchase_deposit_product_id.id
+        deposit = self.product_model.browse(deposit_id)
```

Comment:
Same for other parts in this file where the same lines are repeated.

```suggestion
        deposit = self.env.company.purchase_deposit_product_id
```